### ISSUE TITLE
Implement quantum Louvain algorithms by Cade et al

### ISSUE THREAD

**Issue by timbittner** [qubrabench/qubrabench#18]
Depends on #17.

- [ ] QLouvain (Search + MaxFind, just Search)
- [ ] Repeat Version of QLouvain (Search + MaxFind, just Search)
- [ ] QEdgeLouvain: grover/max on edges (also implement a classical version)
- [ ] #96

**Comment by catch22** [comment]
What's the situation here - on hold until the classical implementation is validated?

**Comment by timbittner** [comment]
The scaffold implementation is completed. We now have to see how to access nested data (pass `max` results out to the scope that called the enclosing `search`, for example) and then validate benchmarking results
